sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnSignaturesTest.java

```diff
@@ -263,4 +392,15 @@ public void foo(ProcessContext context) {}
     @ProcessElement
     public void foo(ProcessContext context) {}
   }
+
+  private static class DoFnForTestSimpleTimerIdNamedDoFn extends DoFn<KV<String, Integer>, Long> {
+    @TimerId("foo")
+    private final TimerSpec bizzle = TimerSpecs.timer(TimeDomain.EVENT_TIME);
+
+    @ProcessElement
+    public void foo(ProcessContext context) {}
+
+    @OnTimer("foo")
+    public void onFoo() {}
+  }
```

Comment:
Should also have a test where the StateSpec / TimerSpec are defined in a parent class.

As a follow-up, should we require they be protected to access them in a child class? Technically we don't care, since they're accessed via the `@TimerId` annotation and reflection, but it may be good to have similar scoping?